sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeCustomizationTest.java

```diff
@@ -50,14 +53,137 @@
+        restartProcessAndSdk();
+
+        /* Set Distribute listener and customize it. */
+        DistributeListener listener = mock(DistributeListener.class);
+        Distribute.setListener(listener);
+
+        /* Resume activity. */
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* Verify the right listener gets called. */
+        verify(listener).onNoReleaseAvailable(mActivity);
+        verify(listener, times(0)).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
+
+        /* Resume activity again to invoke update request. */
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* Verify the right listener gets called again. */
+        verify(listener).onNoReleaseAvailable(mActivity);
+        verify(listener, times(0)).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
+    }
+
+    @Test
+    public void distributeListenerNoReleaseForUser() throws Exception {
+
+        /* Mock error parsing. */
+        ErrorDetails errorDetails = mock(ErrorDetails.class);
+        when(errorDetails.getCode()).thenReturn(ErrorDetails.NO_RELEASES_FOR_USER_CODE);
+        mockStatic(ErrorDetails.class);
+        when(ErrorDetails.parse(anyString())).thenReturn(errorDetails);
+
+        /* Mock http call. */
+        mockHtpCallFailed("{code:'no_releases_for_user'}");
+
+        /* Start Distribute service. */
+        restartProcessAndSdk();
+
+        /* Set Distribute listener and customize it. */
+        DistributeListener listener = mock(DistributeListener.class);
+        Distribute.setListener(listener);
+
+        /* Resume activity. */
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* Verify the right listener gets called. */
+        verify(listener).onNoReleaseAvailable(mActivity);
+        verify(listener, times(0)).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
+    }
+
+    @Test
+    public void distributeListenerNoReleaseFound() throws Exception {
+
+        /* Mock error parsing. */
+        ErrorDetails errorDetails = mock(ErrorDetails.class);
+        when(errorDetails.getCode()).thenReturn(ErrorDetails.NO_RELEASES_FOUND);
+        mockStatic(ErrorDetails.class);
+        when(ErrorDetails.parse(anyString())).thenReturn(errorDetails);
+
+        /* Mock http call. */
+        mockHtpCallFailed("{code:'not_found'}");
+
+        /* Start Distribute service. */
+        restartProcessAndSdk();
+
+        /* Set Distribute listener and customize it. */
+        DistributeListener listener = mock(DistributeListener.class);
+        Distribute.setListener(listener);
+
+        /* Resume activity. */
+        Distribute.getInstance().onActivityResumed(mActivity);
+
+        /* Verify the right listener gets called. */
+        verify(listener).onNoReleaseAvailable(mActivity);
+        verify(listener, times(0)).onReleaseAvailable(any(Activity.class), any(ReleaseDetails.class));
```

Comment:
Please replace `times(0)` to `never()` like everywhere in the project, it does the same, but more readable.